scripts/install-ecs-logs-collector.sh

```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env bash
+set -exo pipefail
+
+# ECS Logs Collector Installation Script
+# Downloads and installs the AWS ECS logs collector from the official GitHub repository
+# Pins to specific script version using commit hash
+
+# Configuration
+readonly INSTALLATION_DIR="/opt/amazon/ecs"
+readonly VERSION_FILE="ECS_LOG_COLLECTOR_VERSION"
+readonly SCRIPT_FILE="ecs-logs-collector.sh"
+
+# Create directory for optional shell scripts
+sudo mkdir -p ${INSTALLATION_DIR}
+
+# Set appropriate file permissions
+sudo chmod 755 ${INSTALLATION_DIR}
+
+# Download ecs-logs-collector.sh
+echo "Downloading ${SCRIPT_FILE} with commit hash: ${ECS_LOGS_COLLECTOR_COMMIT_HASH}"
+sudo curl -fsSL -O --output-dir ${INSTALLATION_DIR} https://raw.githubusercontent.com/aws/amazon-ecs-logs-collector/${ECS_LOGS_COLLECTOR_COMMIT_HASH}/${SCRIPT_FILE}
```

Comment:
This whole script will probably need to be skipped for AIR_GAPPED regions, since it wont have access to github. This might also be a problem sometimes in aws-cn and aws-us-gov partitions. Have we considered getting rid of the logs collector in a separate repo and just maintain it here, so that we're not adding a dependency on an http call to github into our AMI build process?

Reply:
I had wondered and learned a bit more about that issue. I think we can take a different approach using git submodules with a spare checkout to pull in just the ecs-logs-collector.sh script to this repository. The existing provision script copies the script (replaces curl) and the rest remains (minus the README bloat).

That should address AIR_GAPPED and problematic partitions as it should be present in the repository at build time